### ISSUE TITLE
LightGBM's original predictions differ from the transpiled JS predictions

### ISSUE THREAD

**Issue by candalfigomoro** [BayesWitnesses/m2cgen#218]
I've a LGBMRegressor model (LightGBM v2.3.1) and I transpiled it to JavaScript code.

My dataset has about 60 numeric features (no categorical features). Features can contain missing values (they are `np.nan` in Python code and `null` in JavaScript).

This is my LightGBM model:
```
lgb.LGBMRegressor(
    objective='mse',
    max_depth=5,
    first_metric_only=True,
    boosting_type='gbdt',
    importance_type='gain',
    feature_fraction=np.sqrt(len(features))/len(features),
    subsample=0.4,
    seed=RANDOM_STATE,
    eta=0.02,
    nthread=0,
    reg_alpha=0.1,
    reg_lambda=0.1,
    num_leaves=31,
    n_estimators=50
) 
```

Unfortunately I can't share the dataset.

The problem is that LightGBM's original predictions (from Python) differ from the transpiled JS predictions. They are similar, but different. If I plot a distribution of the differences between the original predictions and the JS predictions, I get a sort of normal distribution with mean=0, but in the tail of the distribution there are some pretty large differences.

**Comment by izeigerman** [comment]
Hey @candalfigomoro, thanks for reporting this and sharing your model's hyperparameters! We'll be trying to reproduce this soon and will then come back to you.

**Comment by StrikerRUS** [comment]
Hello @candalfigomoro !
We've tried to reproduce the issue with random data but unfortunately without any success. Since you unable share the original dataset, maybe you can share the trained pickled model? It will help us a lot in debugging the issue.

**Comment by StrikerRUS** [comment]
@candalfigomoro 

> maybe you can share the trained pickled model

If this is not an option too, then please do the following. I have a guess that it may be an issue with the original LightGBM model. Please dump your model to JSON 
```
your_model.booster_.dump_model()
```
and do search for any strange things, e.g. unsupported in JSON values like `nan`/`inf`.

**Comment by StrikerRUS** [comment]
ping @candalfigomoro

**Comment by StrikerRUS** [comment]
Should be fixed in #290. Feel free to reopen if the problem is still persist.
